File: Ej2/src/impl/carrera_carretera.py
from src.carrera import Carrera


class CarreraCarretera(Carrera):
    def __init__(self, bicicletas=[]):
        super().__init__(bicicletas=bicicletas, tipo='Carretera')

    def iniciar_carrera(self):
        print('Iniciando carrera de carretera')
        
        for i in range(self.__bicicletas * 0.1):
            bicicleta = self.__retirar_bicicleta_aleatoria()
            print(f'La bicicleta {bicicleta.get_identificador()} ha terminado la carrera')

File: Ej1/src/practica1ds/CarreraCarretera.java
package practica1ds;

public class CarreraCarretera extends Carrera {

    public CarreraCarretera(String tipo) {
        super(tipo);
    }

    @Override
    public void run() {
        try {
            System.out.println("Empieza la Carrera de carretera con " + bicicletas.size() + " bicicletas\n");
            Thread.sleep(60000);
            for (int i = 0; i < bicicletas.size() * 0.10; i++) {
                retirarBicicletaAleatoria();
            }

        } catch (InterruptedException e) {

            e.printStackTrace();
        }
    }
}
